### PR TITLE
Add Test suite

### PR DESCRIPTION
**Pull request by daquinons**
# Description

This request installs these packages:
- [React Testing Library](https://testing-library.com/docs/react-testing-library/intro)
- [Jest Dom](https://github.com/testing-library/jest-dom)
- [Prop-Types](https://github.com/facebook/prop-types)

It also updates the App test to use react-testing-library in order to render the `<App />`, so after this merge request all the tests will be required to use these libraries

## Checklist

Remove any items which are not applicable.

- [X] I have performed a self-review of my own code
- [X] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works AND the tests pass

### CODE REVIEW

**Comment by daquinons** [comment]
[Set up test suite (Frontend)](https://trello.com/c/L7v3sVl8/20-set-up-test-suite-frontend)

File: src/App.test.js [hunk 1]
```diff
@@ -1,9 +1,8 @@
...
-import ReactDOM from 'react-dom';
+import { render } from '@testing-library/react';
+import '@testing-library/jest-dom/extend-expect';
 import App from './App';
 
 it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
+  render(<App />);
```
Comment (AlexEntrepreneur) [inline src/App.test.js]: Could create an assertion to test that `jest-dom` is working correctly.

Comment (daquinons) [reply src/App.test.js]: Good idea, thanks Alex 😊 I just pushed the test using it.

File: src/App.test.js [hunk 2]
```diff
@@ -1,9 +1,11 @@
...
+import '@testing-library/jest-dom/extend-expect';
 import App from './App';
 
 it('renders without crashing', () => {
-  const div = document.createElement('div');
-  ReactDOM.render(<App />, div);
-  ReactDOM.unmountComponentAtNode(div);
+  const { getByTestId } = render(<App />);
+  const AppComponent = getByTestId('App');
+  expect(AppComponent).toBeInTheDocument();
```
Comment (Pav0l) [inline src/App.test.js]: 😍

**Pull request merged**